src/chatlog/chatmessage.cpp

```diff
@@ -184,8 +184,11 @@ QString ChatMessage::detectAnchors(const QString &str)
                 "(?:\\b)(www\\.|http[s]?|ftp)://" // (protocol)://(printable - non-special character)
                 // http://ONEORMOREALHPA-DIGIT
                 "\\w+\\S+)" // any other character, lets domains and other
-                "|(^tox:[@\\w]+$)"); // starts with `tox` then : and only alpha-digits till the end
-                // also accepts tox:agilob@net as simplified TOX ID
+                "|"   // standard protocols OR tox ID:
+                "^tox:(([\\w]+@[\\S]+)" // tox:something@something.com
+                "|(\\w{76}))$" // tox:76 characters long tox id
```

Comment:
`\\S+` is any printable character [A-Za-z0-9]+ special characters
`\\w+` is [A-Za-z0-9]

in fact it should be [A-Fa-f0-9] but that would be much slower, keep in mind this regexp is run on every message and action.